docs_src/design/ucr/URIs-for-Files.md

```diff
@@ -0,0 +1,60 @@
+
+    - Location of this file is specified in configuration [here](https://github.com/edgexfoundry/edgex-go/blob/v2.3.0/cmd/core-metadata/res/configuration.toml#L50) 
+
+- Service Configuration files
+    - Location of these files are currently defaulted to be `./res/configuration.toml`, but can be overridden  via -cf/--configFile command line flag.
+    - The Common Configuration ADR adds a new [common configuration file](https://docs.edgexfoundry.org/3.0/design/adr/0026-Common%20Configuration/#specifying-the-common-configuration-location) specified via the future -cc/--commonConfig command line flag.
+
+- [Token Configuration](https://github.com/edgexfoundry/edgex-go/blob/v2.3.0/cmd/security-file-token-provider/res/token-config.json) file for Security File Token Provider 
+
+    - This file specifies the list of services to generate Vault tokens for. 
+
+- Device Profiles, Device Definition and Provision Watchers
+
+    - These files can reside in a device services local file system and are pushed to Core Metadata the first time the service starts. Example [here](https://github.com/edgexfoundry/device-onvif-camera/tree/v2.3.0/cmd/res)
+
+    - These files are found by scanning the folders specified in configuration [here](https://github.com/edgexfoundry/device-sdk-go/blob/main/example/cmd/device-simple/res/configuration.toml#L94-L96)
+
+    !!! note 
+        These files are only pushed to Core Metadata the first time the device service is loaded. They are not currently re-pushed once they exist in Core Metadata even when the files have changed locally. Thus updating the files locally or in a shared location will not result in changing the contents of these files in Core Metadata. They still benefit from this capability during initial deployment and when new files are added.  
+
+
+Currently all files loaded by services are expected to be on the local file system, thus are duplicated many times when deploying at scale.
+
+### Target Users
+- Software Deployer
+- Software Integrator
+
+### Description
+This UCR proposes to enhance loading of files in EdgeX by allowing the location of the file to be optionally specified as an URI.
+
+### Existing solutions
+Loading shared files via a URI is not new in the software industry. Here is the Wiki page for [Uniform Resource Identifier](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier) 
+
+### Requirements
+- Single EdgeX service files shall optionally be loaded via a specified URI. 
+- Sets of EdgeX service files (i.e. device service files) shall optionally be loaded via a URI. Details on how are left to the ADR.
+- The URIs specified shall follow the [Uniform Resource Identifier](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier) spec when authentication not required or using **basic-auth** in plain text in the URI,  i.e. `username:password@`
+  - Only the `http`, `https`, and `file` schemes from the above spec shall be supported.
```

Comment:
I assume that the loading of local files via plain paths remains supported as the [File URI](https://en.wikipedia.org/wiki/File_URI_scheme) does not support relative paths. The implementation can simply convert a file URI to a local path (or vice versa) and process it as done right now. 

That makes me wonder the usefulness of additionally supporting the File URI when in all situations covered in this UCR, the existing design requires the support for plain file paths (absolute and relative). There is really no harm in having it as it shows extensibility: one may add sFTP to securely read files over SSH with key-based authentication from a remote server. It does add unnecessary complexity in code and docs.

Reply:
Good point. Yes we need to support "plain paths". Will update to replace `file` with `plain paths`. As for sFTP, not a fan due to as you stated "It does add unnecessary complexity ".

Reply:
Update doc

Reply:
That should work.
I meant keeping the File URI support adds unnecessary complexity. I wasn't suggesting the addition of sFTP in reference implementation.

Reply:
That should work.
I meant keeping the File URI support adds unnecessary complexity. I wasn't suggesting the addition of sFTP in reference implementation.